apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java

```diff
@@ -53,107 +42,100 @@
 import static co.elastic.apm.agent.servlet.ServletTransactionHelper.TRANSACTION_ATTRIBUTE;
 import static co.elastic.apm.agent.servlet.ServletTransactionHelper.determineServiceName;
 
-public class ServletApiAdvice {
+public abstract class ServletApiAdvice {
 
     private static final String FRAMEWORK_NAME = "Servlet API";
     static final String SPAN_TYPE = "servlet";
     static final String SPAN_SUBTYPE = "request-dispatcher";
     private static final ServletTransactionHelper servletTransactionHelper;
-    private static final ServletTransactionCreationHelper servletTransactionCreationHelper;
 
     static {
         servletTransactionHelper = new ServletTransactionHelper(GlobalTracer.requireTracerImpl());
-        servletTransactionCreationHelper = new ServletTransactionCreationHelper(GlobalTracer.requireTracerImpl());
     }
 
     private static final GlobalThreadLocal<Boolean> excluded = GlobalThreadLocal.get(ServletApiAdvice.class, "excluded");
     private static final GlobalThreadLocal<Object> servletPathTL = GlobalThreadLocal.get(ServletApiAdvice.class, "servletPath");
     private static final GlobalThreadLocal<Object> pathInfoTL = GlobalThreadLocal.get(ServletApiAdvice.class, "pathInfo");
 
-    private static final List<String> requestExceptionAttributes = Arrays.asList(RequestDispatcher.ERROR_EXCEPTION, "exception", "org.springframework.web.servlet.DispatcherServlet.EXCEPTION", "co.elastic.apm.exception");
+    private static final List<String> requestExceptionAttributes = Arrays.asList("javax.servlet.error.exception", "jakarta.servlet.error.exception", "exception", "org.springframework.web.servlet.DispatcherServlet.EXCEPTION", "co.elastic.apm.exception");
 
     @Nullable
-    @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
-    public static Object onEnterServletService(@Advice.Argument(0) ServletRequest servletRequest) {
+    public static <ServletRequest, ServletResponse, HttpServletRequest, HttpServletResponse, ServletContext> Object onServletEnter(ServletRequest servletRequest, ServletHelper<ServletRequest, ServletResponse, HttpServletRequest, HttpServletResponse, ServletContext> helper) {
```

Comment:
The generic type definitions using classnames are confusing, and would be even more confusing if any of these were imported to be used somewhere in a later change. I suggest changing them to uppercase, eg ServletRequest -> REQUEST, HttpServletRequest-> HTTPREQUEST etc

Reply:
fixed